### PR TITLE
Do an mdns lookup when KMS url ends in .local

### PR DESCRIPTION
**Pull request by jimmykarily**
Part of: https://github.com/kairos-io/kairos/issues/2069

### CODE REVIEW

File: go.mod
Comment (jimmykarily) [reply go.mod:6]: ok removed all other changes now. Waiting for the tpm-helpers PR to be merged.

File: mdns-notes.md
Comment (jimmykarily) [inline mdns-notes.md:1]: I will move this to kairos-docs as soon as we release

**Comment by jimmykarily** [comment]
I skipped the new test for now to unblock the release. I will create a ticket to fix it post-release. Manual testing works.

File: Earthfile
Comment (mauromorales) [inline Earthfile:4]: I think you should be able to do something like:

https://github.com/kairos-io/kairos/blob/master/Earthfile#L11

and 

https://github.com/kairos-io/kairos/blob/master/renovate.json#L24

but you will need to extend the flavor_release part.

Maybe we should talk about introducing some other tags, e.g. this could be something like `ubuntu:rolling-core-amd64-generic-stable` or so

**Review by mauromorales — APPROVED** [review]
great job man :clap:

File: Earthfile
Comment (jimmykarily) [reply Earthfile]: Nice! I add the renovate things. Should be ok for now.

File: mdns-notes.md
Comment (jimmykarily) [reply mdns-notes.md:1]: https://github.com/kairos-io/kairos-docs/pull/134

**Comment by jimmykarily** [comment]
Issue created for the skipped test: https://github.com/kairos-io/kairos/issues/2182

**Pull request merged**
